tensorzero-internal/src/endpoints/object_storage.rs

```diff
@@ -0,0 +1,62 @@
+use axum::{
+    extract::{Query, State},
+    Json,
+};
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    config_parser::ObjectStoreInfo,
+    error::{Error, ErrorDetails},
+    gateway_util::{AppState, AppStateData},
+    inference::types::storage::StoragePath,
+};
+use aws_smithy_types::base64;
+
+#[derive(Debug, Serialize)]
+pub struct ObjectResponse {
+    data: String,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct PathParams {
+    storage_path: String,
+}
+
+pub async fn get_object_handler(
+    State(AppStateData { .. }): AppState,
+    Query(params): Query<PathParams>,
+) -> Result<Json<ObjectResponse>, Error> {
+    // TODO - should we re-use the config object store if it matches?
```

Comment:
I think this is worth doing. Otherwise on a page with many images this is going to initialize a bunch of object store clients which I assume contain connection pools.